src/components/CardGrid/CardGrid.tsx

```diff
@@ -51,13 +53,14 @@ function CardGrid({
   const breakpoint: Breakpoint = useBreakpoint();
   const isLargeScreen = breakpoint >= Breakpoint.md;
   const imageSourceWidth = 320 * (window.devicePixelRatio > 1 || isLargeScreen ? 2 : 1);
-  const rows = chunk<PlaylistItem>(playlist, cols[breakpoint]);
+  const rows = chunk<PlaylistItem>(playlist.playlist, cols[breakpoint]);
 
   const cellRenderer = ({ columnIndex, rowIndex, style }: GridCellProps) => {
     if (!rows[rowIndex][columnIndex]) return;
 
     const playlistItem: PlaylistItem = rows[rowIndex][columnIndex];
     const { mediaid, title, duration, seriesId, episodeNumber, seasonNumber } = playlistItem;
+    const [image, fallbackImage] = getCardImages(playlistItem, playlist, imageSourceWidth);
```

Comment:
It's also a little bit weird how we get the images in the component instead of when the data is fetched (like you did with epg channels.)  It might be nice to calculate these once in api.service.ts and attach them to the items there (the Playlist format returned from the api should ideally be more decoupled from the type used in components, but that may be too much for the moment.)

Reply:
Done